k2/csrc/ragged_ops.cu

```diff
@@ -1143,7 +1144,26 @@ Array1<int32_t> GetTransposeReordering(Ragged<int32_t> &src, int32_t num_cols) {
   if (device_type == kCpu) return GetTransposeReorderingCpu(src, num_cols);
 
   K2_CHECK_EQ(device_type, kCuda);
+  (void)GetTransposeReorderingThreeAxesCuda;  // remove compiler warnings
 
+#if 1
+  const int32_t *values_data = src.values.Data();
+  int32_t num_buckets = num_cols;
+  int32_t num_elements = src.values.Dim();
+
+  Array1<int32_t> order = Range(context, num_elements, 0);
+  Array1<int32_t> ans(context, num_elements);
+
+  auto bucket_mapping_lambda =
+      [values_data] __device__(uint32_t i) -> uint32_t {
+    return values_data[i];
+  };
+
+  MultiSplitKeysOnly(context, num_elements, num_buckets, &bucket_mapping_lambda,
+                     reinterpret_cast<const uint32_t *>(order.Data()),
```

Comment:
How about we just pass `values_data` as the keys and pass order as the values to call `MultiSplitKeyValuePaires`? No extra memory allocation compared to current approach but saving a mapping lambda?

Reply:
I think we want to get a permutation of [0, 1, 2, 3, 4, ..., N].

If we pass `values_data`, the returned value would be a permutation of `values_data`.

Reply:
No, I mean, you pass `value_data` as the keys (as you did with `bucket_mapping_lambda`?) and pass `order = Range(context, num_elements, 0) ` as `values` (`ans` here) to call `SortKeyValuePairs`

Reply:
I use RadixSort directly in this case.